subsys/bluetooth/controller/ll_sw/ull_adv_sync.c

```diff
@@ -586,6 +614,236 @@ static inline uint8_t sync_remove(struct ll_adv_sync_set *sync,
 	return 0U;
 }
 
+/* @brief Set or clear fields in extended advertising header.
+ *
+ * @param[in]  adv              Advertising set.
+ * @param[in]  hdr_add_fields   Flag with information which fields add.
+ * @param[in]  hdr_rem_fields   Flag with information which fields remove.
+ * @param[in]  value            Pointer to data to be added to header.
+ *                              Content depends on the value of
+ *                              @p hdr_add_fields.
+ * @param[out] ter_idx          Index of new PDU.
+ *
+ * @Note
+ * @p value depends on the flag provided by @p hdr_add_fields.
+ * Information about content of value may be found in description of
+ * @ref ull_adv_sync_pdu_set_clear.
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+static uint8_t adv_sync_hdr_set_clear(struct lll_adv_sync *lll_sync,
+				      struct pdu_adv *ter_pdu_prev,
+				      struct pdu_adv *ter_pdu,
+				      uint16_t hdr_add_fields,
+				      uint16_t hdr_rem_fields,
+				      void *value)
```

Comment:
I did it this way because I wanted to avoid commit altering changes in previous commit.
May I consider the suggestion about separate commit as a hint for future?

Reply:
Sure - It'll be a pain to separate them now :D